### PR TITLE
Improve policy translation API

### PR DESCRIPTION
**Pull request by csilvestrini**
Now returns additional checks/claims rather than mutating a RecipeGraph. Reverted previous changes to RecipeGraph where claims/checks could be added to nodes -- mutating a graph was more subtle and error prone than that...

### CODE REVIEW

File: java/arcs/core/analysis/RecipeGraph.kt [hunk 1]
```diff
@@ -95,17 +93,7 @@ data class RecipeGraph(
...
-            val claims: List<Claim>,
-            val checks: List<Check>
-        ) : Node() {
-            constructor(particle: Recipe.Particle) : this(
-                particle,
-                particle.spec.claims,
-                particle.spec.checks
-            )
-
+        data class Particle(val particle: Recipe.Particle) : Node() {
```
Comment (bgogul) [inline java/arcs/core/analysis/RecipeGraph.kt:96]: May be also revert the changes to the constructor of `RecipeGraph` or at least make it private? We should not allow constructing a `RecipeGraph` directly from `particleNodes` and `handleNodes`.

Comment (csilvestrini) [reply java/arcs/core/analysis/RecipeGraph.kt:95]: Completely forgot about that, thanks
